token/js/src/instructions/initializeImmutableOwner.ts

```diff
@@ -0,0 +1,128 @@
+import { struct, u8 } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import { AccountMeta, PublicKey, TransactionInstruction } from '@solana/web3.js';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+} from '../errors';
+import { TokenInstruction } from './types';
+
+/** TODO: docs */
+export interface InitializeImmutableOwnerInstructionData {
+    instruction: TokenInstruction.InitializeImmutableOwner;
+}
+
+/** TODO: docs */
+export const initializeImmutableOwnerInstructionData = struct<InitializeImmutableOwnerInstructionData>([
+    u8('instruction'),
+    u8('immutableOwnerOption'),
+]);
+
+/**
+ * Construct an InitializeImmutableOwner instruction
+ *
+ * @param account            Account
+ * @param immutableOwner  Optional authority that can close the mint
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createInitializeImmutableOwnerInstruction(
+    account: PublicKey,
+    immutableOwner: PublicKey | null,
```

Comment:
This parameter is unnecessary, since the `immutableOwner` is just the owner :smile: